llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -1131,6 +1191,92 @@ Operation *ReverseOp::getTiledImplementation(OpBuilder &builder,
   return tiledRevOp;
 }
 
+//===----------------------------------------------------------------------===//
+// BinCountOp
+//===----------------------------------------------------------------------===//
+
+static LogicalResult verifyBinCountOp(BinCountOp op) {
+  int64_t numInputs = op.getNumInputs();
+  if (numInputs < 1 || numInputs > 2) {
+    return op.emitOpError("expected atleast one and atmost two input operands");
+  }
+  if (op.getNumOutputs() != 1) {
+    return op.emitOpError("expected one output operand");
+  }
+  auto inputType = op.input().getType().cast<ShapedType>();
+  ArrayRef<int64_t> inputShapes = inputType.getShape();
+  if (inputShapes.size() != 1) {
+    return op.emitError("expected bins rank to be one");
+  }
+  if (!inputType.getElementType().isa<mlir::IntegerType>()) {
+    return op.emitError("expected bins element type to be integer");
+  }
+  // Check if the number of input operands is two, which means weights are also
+  // given as input.
+  if (numInputs == 2) {
+    auto weightsType = op.weights().getType().cast<ShapedType>();
+    ArrayRef<int64_t> weightsShapes = weightsType.getShape();
+    if (weightsShapes.size() != 1) {
+      return op.emitError("expected weights rank to be one");
+    }
+    if (inputShapes[0] != weightsShapes[0]) {
+      return op.emitOpError("incompatible bins and weights shapes");
+    }
+    auto outputType = op.output().getType().cast<ShapedType>();
+    // If weights are given as input, the output must be floating-point type.
```

Comment:
Shouldn't the element type of output be the same as weights? So if weights are i32, output i32? And if weights are f32, output is f32?

Reply:
I checked this in python. When weights are given as input, the output is always a floating-point, irrespective of the type of weight.

Reply:
Makes sense to stick with torch's conventions. Looks good!